lib/Analysis/Membar.cpp

```diff
@@ -138,14 +138,19 @@ void MembarAnalysis::update(Operation *op, BlockInfo *blockInfo,
     for (Value value : op->getOperands()) {
       for (auto bufferId : allocation->getBufferIds(value)) {
         if (bufferId != Allocation::InvalidBufferId) {
-          if (isa<triton::gpu::AsyncCopyGlobalToLocalOp>(op)) {
-            // Global -> shared memory
-            curBlockInfo.syncWriteIntervals.insert(
-                allocation->getAllocatedInterval(bufferId));
-          } else {
-            // ConvertLayoutOp: shared memory -> registers
-            curBlockInfo.syncReadIntervals.insert(
-                allocation->getAllocatedInterval(bufferId));
+          MemoryEffectOpInterface memIF(op);
```

Comment:
This could probably be tested blindly on all ops and if they have Read or Write on a Shared resource type, then add the intervals (but I guess the buffer ID would still be needed).

Reply:
There probably needs some deeper cleaning.

I think you could use this interface:

```
  EffectInstance(EffectT *effect, Value value,
                 Resource *resource = DefaultResource::get())
```